iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/StringNoSpacing.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.api;
+
+public interface StringNoSpacing extends DockerTree {
```

Comment:
Instead of a class per kind of Strings, you could implement the StringLiteral interface and describe string types/kinds. In the end, we are only interested in the value. In addition, there should not be a difference between StringNoSpacing and StringWithSpace. It is only a grammar specification but not relevant to the AST. Creating so many different types of trees will make it hard to handle them.